### DISCUSSION TITLE
C# Client Code Generation. Include "Security"

### DISCUSSION THREAD

**Discussion by damiensprinkle** [RicoSuter/NSwag#3948] [Q&A]
Hello, I'm curious if there is anyway for the generated c# client code to include security schemas

Example from endpoint,

 "security": [
          {
            "oauth2": [
              "Internal"
            ]
          }
          
          
Our endpoints use role based authorization, I was hoping to have this at least visible in the generated code.

**Comment by rvilchesRA** [comment]
I imagine that a Basic authorization (user-password).
I modified the client and added the following.

var byteAuth = Encoding.ASCII.GetBytes("user:passord");
request_.Headers.Authorization = new AuthenticationHeaderValue("Basic", Convert.ToBase64String(byteAuth));
